p4-16/psa/PSA.mdk

```diff
@@ -467,19 +472,21 @@ assist P4 developers in debugging their code.
     }
     if (platform_port_valid(ostd.egress_port)) {
         enqueue one packet for output port ostd.egress_port with class
-            of service specified by ostd.class_of_service
+            of service ostd.class_of_service
     } else {
-        drop the packet.
+        drop the packet
         // Recommended to log error about unsupported ostd.egress_port
         // value.
     }
 ```
 
-TBD: Need text defining, for each possible copy, exactly what the
-contents of the packet will be.
+TBD: Should it be possible to truncate a cloned packet differently
+than the normal packet that goes out?
 
-TBD: Should it be possible to truncate a cloned or resubmitted packet
-differently than the normal packet that goes out?
+TBD: The definition of `ostd.truncate_payload_bytes` refers to payload
+bytes.  This makes sense for normal packets passing through the
+ingress deparser, but what does it mean for clone packets, where the
+packet is supposed to be the original pre-IngressParser packet.
```

Comment:
If you know a good answer for this, let me know.  One possibility: have a separate metadata value output by ingress control that specifies the truncation length for CLONE_I2E packets independently from the truncation length of normal packets.

Reply:
Shall we just disallow truncation of cloned packets?

Reply:
I would think that perhaps one common situation in which truncating packets is desired is when sending a clone to the CPU -- the CPU software might often only care about the headers, not the payload contents.

Reply:
One way to specify this would be to have a clone_truncate_length metadata field, like we have other clone-specific metadata fields.  For ingress, it would specify the maximum number of packet bytes to be cloned, starting at the beginning of the first header byte, not payload bytes.  For egress, it would specify the maximum number of payload bytes, but independently from packet truncation of normal packets, so a P4 programmer could make a clone that is truncated, and a normal packet that is not, in the same egress processing pass of a single packet.

Reply:
Wouldn't it be easier to create a digest?

Reply:
I have no problem saying in the PSA that truncation is not supported for clone packets, and suggesting the use of the Digest extern if/when someone wishes for such functionality.  If someone wants truncation of cloned packets, we can bring it up in the PSA WG in the future.